pkg/controller/installation/validation.go

```diff
@@ -260,6 +264,11 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 			if instance.Spec.CNI.Type != operatorv1.PluginCalico {
 				return fmt.Errorf("spec.calicoNetwork.hostPorts is supported only for Calico CNI")
 			}
+
+			if instance.Spec.CalicoNetwork.LinuxDataplane != nil && *instance.Spec.CalicoNetwork.LinuxDataplane == operatorv1.LinuxDataplaneBPF {
```

Comment:
Presumably if host ports is explicitly "Disabled" that would be an acceptable config?

Reply:
I guess, but the check above didn't seem to make that exception?  Happy to fix up both if that makes sense

Reply:
I think the difference is that if the CNI plugin isn't Calico, we actually can't control if host ports is enabled or disabled, since we're not actually installing and configuring the CNI plugin, so accepting either option (Enabled or Disabled) would be wrong, since it's out of our control.

If the CNI plugin is Calico, but the data plane mode is BPF, we still control whether the hostport plugin is actually installed / configured, so we could choose to accept the values for that field that are compatible. 

It's a minor thing though. Worth checking if we actually configure hostPorts if that field is nil, though.

Reply:
Maybe it's different for non-Calico CNI; the validation there is saying "we're not even in control so even Disabled makes no sense".

Reply:
Crossed in the post(!) I adjusted it to ignore disabled only for BPF mode.